addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -485,16 +486,20 @@ def drop_off_picked(self, continue_batch, move_line_ids, location_barcode):
 
             pickings = completed_move_lines.mapped('picking_id')
 
+            to_add = Picking.browse()
             for pick in pickings:
                 pick_todo = pick
                 pick_mls = completed_move_lines.filtered(
                     lambda x: x.picking_id == pick)
 
                 if pick._requires_backorder(pick_mls):
                     pick_todo = pick._backorder_movelines(pick_mls)
+                    to_add |= pick_todo
 
                 # at this point pick_todo should contain only mls done
                 pick_todo.update_picking(validate=True)
+            # Add backorders to the batch
```

Comment:
Style nitpick: add an empty line above please, to better separate the `for` block